dg/tests.md

```diff
@@ -0,0 +1,166 @@
+<frontmatter>
+  header: header.md
+  pageNav: 2
+  pageNavTitle: "Chapters of This Page"
+  siteNav: dg-nav.md
+  footer: footer.md
+</frontmatter>
+
+# Introduction to Tests in CATcher
+
+1. [Jasmine Style Guide](#jasmine-style-guide)
+    1. [Test Suite](#test-suite)
+    2. [Spy](#spy)
+    3. [Resources for Jasmine](#resources-for-jasmine)
+2. [Angular TestBed Utility](#angular-testbed-utility)
+    1. [Resources for Angular TestBed](#resources-for-angular-testbed)
+3. [E2E Testing](#e2e-testing)
+    1. [Running E2E Tests](#running-e2e-tests)
+    2. [Troubleshooting conflicts between the versions of the browser and browser driver](#troubleshooting-conflicts-between-the-versions-of-the-browser-and-browser-driver)
+    3. [Protractor Configuration](#protractor-configuration)
+    4. [How the E2E tests work](#how-the-e2e-tests-work)
+
+## Jasmine Style Guide
+
+Jasmine is a behavior-driven development framework specific for testing JavaScript code. We follow the Jasmine Style Guide loosely for our tests.
+
+### Test Suite
```

Comment:
In my view, there is no need to describe how Jasmine or its various features are used (unless there is some non-trivial information specific to CATcher, as seen in the case of the E2E tests).

For each tool used by our project, it should be sufficient to explain *what* the tool is / *why* is it needed in the project.

If we include detailed info on how to use the tool / how it works, we'll simply be replicating information already provided on the web by experts. We'll also have to ensure that the info stays updated.

Hence, let's remove the explanation provided here for Jasmine (the `Test Suite` and `Spy` sections).

Reply:
Sure, I have removed it!